### PR TITLE
(7.0) Refactor robotest configuration - keep upgrade_from releases current.

### PR DESCRIPTION
**Pull request by wadells**
## Description
Port robotest config refactoring from #1837 .

`pr_config.sh` is the only file with changes of interest to this branch -- `pr_nightly.sh` has branch specific changes but they'll never be run.

I removed test.sh and references to it since it [may still be in flux](https://github.com/gravitational/gravity/pull/1837#discussion_r453121148).

## Type of change
* Internal change (not necessarily a bug fix or a new feature)

## Linked tickets and other PRs
* Ports #1837 

## TODOs
- [x] Self-review the change
- [x] Audit automated test run for correctness
- [ ] Address review feedback

## Testing done
See  #1837 for the original testing. For 7.0: the PR Jenkins run should sufficiently exercise the 7.0 PR build relevant changes. We don't care about testing the nightly changes because we don't run nighties on 7.0 so although I made an minor effort to keep this file branch relevant mistakes are moot.

### CODE REVIEW

File: build.assets/robotest/nightly_config.sh [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+set -o nounset
+set -o pipefail
+
+source $(dirname $0)/utils.sh
+
+# UPGRADE_MAP maps gravity version -> list of linux distros to upgrade from
+declare -A UPGRADE_MAP
+
+# UPGRADE_MAP maps gravity version -> list of linux distros to upgrade from
+declare -A UPGRADE_MAP
```
Comment (wadells) [inline build.assets/robotest/nightly_config.sh:12]: UPGRADE_MAP is redeclared.

Comment (wadells) [reply build.assets/robotest/nightly_config.sh]: Fixed in bc0399b.

**Review by a-palchikov — APPROVED** [review]

**Comment by wadells** [comment]
All the CI failures observed here are with CentOS:8 runs.  Bootstrap errors that look to stem from `semanage: command not found`.  I've filed https://github.com/gravitational/robotest/issues/258 to get this addressed in robotest.

I'll remove the unsupported configs from this (& other ports) and add some information about how the suites change overall.

**Comment by wadells** [comment]
OSes updated to be safer.  Here is a diff of the test config:

```
walt@work:~/git/gravity$ diff pr.before pr.after                               
5,8c5,9
< upgrade={"flavor":"three","nodes":3,"role":"node","service_uid":997,"service_gid":994,"os":"ubuntu:16","storage_driver":"overlay2","from":"/telekube_6.1.0.tar"}
< upgrade={"flavor":"three","nodes":3,"role":"node","service_uid":997,"service_gid":994,"os":"ubuntu:18","storage_driver":"overlay2","from":"/telekube_6.1.24.tar"}
< upgrade={"flavor":"three","nodes":3,"role":"node","service_uid":997,"service_gid":994,"os":"ubuntu:18","storage_driver":"overlay2","from":"/telekube_6.3.13.tar"}
< upgrade={"flavor":"three","nodes":3,"role":"node","service_uid":997,"service_gid":994,"os":"ubuntu:18","storage_driver":"overlay2","from":"/telekube_7.0.5.tar"}
---
> upgrade={"flavor":"three","nodes":3,"role":"node","os":"centos:7","from":"/telekube_6.1.0.tar","service_uid":997,"service_gid":994,"storage_driver":"overlay2"}
> upgrade={"flavor":"three","nodes":3,"role":"node","os":"debian:9","from":"/telekube_6.3.18.tar","service_uid":997,"service_gid":994,"storage_driver":"overlay2"}
> upgrade={"flavor":"three","nodes":3,"role":"node","os":"redhat:7","from":"/telekube_6.1.31.tar","service_uid":997,"service_gid":994,"storage_driver":"overlay2"}
> upgrade={"flavor":"three","nodes":3,"role":"node","os":"ubuntu:16","from":"/telekube_7.0.0.tar","service_uid":997,"service_gid":994,"storage_driver":"overlay2"}
> upgrade={"flavor":"three","nodes":3,"role":"node","os":"ubuntu:18","from":"/telekube_7.0.12.tar","service_uid":997,"service_gid":994,"storage_driver":"overlay2"}
```

Notably:
* 1 upgrade scenario is added: 7.0.0 -> current
* `centos:7`, `redhat:7` and `debian:9` are mixed into the upgrade matrix.

I'll catch `centos:8`, `redhat:8` and `suse:12` the next time I release robotest, as there are some fixes for those distros in the pipes.  I'm also excited for that, as Dmitri's https://github.com/gravitational/robotest/pull/219 would have greatly sped up our failures observed in the early CI jobs for this PR.

**Review by r0mant — APPROVED** [review]

**Pull request merged**